### PR TITLE
approved partners can make requests

### LINKED ISSUE #8: Approved partner can fill out requests for diapers to their diaper bank partner
We need to create a new form for a partner to fill out so they can request for diapers from their diaper banks. 

The form will have labels to show the current partner's name.
The form will have the following input fields:

```
comments: string,

# The partner can click `Add an item` which will create a new row with 
the following information:

request_item: select dropdown,
quantity: number
```

`Given` the partner has filled out the form
`When` the partner clicks `Submit for Review`
`Then` the system will submit the form information to the diaper app

This will send a post request to the diaper app at `/api/v1/partner_requests`. Here is an example payload:
```json
{
    "request": {
        "partner_id":5,
        "organization_id":1,
        "comments":"Lots and lots of comments",
        "request_items":
            {
            "k_size1":100,
            "k_size2":200,
            "k_size3":200,
            "k_size4":200,
            "k_size5":222,
            "k_size6":444
            }
    }
}
```

### PR DESCRIPTION
**Pull request by chaserx**
Resolves #8 

This PR is a new, requested feature that enables approved/verified partners to submit requests. The partner's request is saved and sent to the diaper app. Currently, the payload of that POST to the diaper app looks like this:

```json
{
  "request": {
    "partner_id": 3,
    "organization_id": 3,
    "comments": "You want to enjoy life, don't you? If you get your job done quickly and your job is fun, that's good isn't it? That's the purpose of life, partly. Your life is better.",
    "request_items": {
      "pullup_23t": "900",
      "cloth": "8"
    }
  }
}

```

**edit**: updated JSON payload to conform to what the diaper application expects.

kapture 2018-10-29 at 7 08 12

**TODO:**
- [x] tests

### CODE REVIEW

**Comment by seanmarcia** [comment]
The payload it sends it a little different than is expected by the diaper app. Let me think about this for a few.

**Comment by chaserx** [comment]
@seanmarcia yeah... i went the easy route on that requirement. i do like how the data is mapped out though.

**Comment by chaserx** [comment]
@seanmarcia removed the WIP. other than the differing payload problem, i think it's good to go.

**Comment by seanmarcia** [comment]
Hey @chaserx I commented in slack on an alternative way you could handle the payload to make it what the diaper app is expecting.

File: app/views/partner_requests/thanks.html.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<h1>Thanks</h1>
```
Comment (seanmarcia) [inline app/views/partner_requests/thanks.html.erb]: should probably be a show page instead of a custom route

File: config/application.rb [hunk 2]
```diff
@@ -15,5 +15,7 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+    config.autoload_paths << Rails.root.join("blueprints")
```
Comment (seanmarcia) [inline config/application.rb]: probably don't need these

File: config/routes.rb [hunk 3]
```diff
@@ -5,6 +5,10 @@
     get :approve
   end
 
+  resources :partner_requests, only: [:new, :create] do
```
Comment (seanmarcia) [inline config/routes.rb]: only: [:new, :create, :show] and get rid of the thanks

Comment (seanmarcia) [reply config/routes.rb]: Should probably have an index page too now that I think about it. They'll probably want to view their old requests.

File: app/controllers/partner_requests_controller.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          @partner_request.errors.add(:base, :sending_failure, message: "request saved but failed to send")
+        end
+        format.html { redirect_to partner_request_thanks_path(@partner_request), notice: "Request was successfully created." }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def thanks
```
[the remark this answers is not in the document]
Comment (chaserx) [reply app/controllers/partner_requests_controller.rb]: yeah. perhaps i was being cute. show is more conventional.

File: app/views/partners/show.html.erb [hunk 5]
```diff
@@ -1,6 +1,24 @@
...
+  <%= @partner.name  %><br>
+  <%= @partner.executive_director_name  %><br>
+  <%= @partner.email  %><br>
+  Status: <%= @partner.partner_status %>
+</p>
 
-<%= link_to 'Edit', edit_partner_path(@partner) %> |
-<%= link_to 'Back', partners_path %>
+<div class="actions">
+  <% if @partner.verified? %>
```
[the remark this answers is not in the document]
Comment (chaserx) [reply app/views/partners/show.html.erb]: Ok. Will do. I moved the `edit` into the nav across the top under the `account` button. I'm not sure if that obscures that action too much or not. 

screen shot 2018-10-30 at 10 03 15 am

**Comment by chaserx** [comment]
Added an index view for Partner Requests

screen shot 2018-10-30 at 10 58 59 am

**Comment by chaserx** [comment]
I think we're good here again.

**Pull request merged**